### PR TITLE
[LLM Testing] Testing configuration

### PR DESCRIPTION
**Pull request by dbogunowicz**

### CODE REVIEW

File: tests/deepsparse/transformers/pipelines/test_text_generation.py [hunk 1]
```diff
@@ -12,285 +12,236 @@
 # See the License for the specific language governing permissions and
 # limitations under the License.
 
-from typing import List, Optional, Tuple
+import os
+from typing import List, Optional, Tuple, Union
```
Comment (dbogunowicz) [inline tests/deepsparse/transformers/pipelines/test_text_generation.py:16]: - model by model (each config has their own file) (stub, model_name, use_bos_token, max_diff).
- cadence
- scenarios: turn off and off the essential files.

Pytest.any to define configs. pytest.ini

File: tests/deepsparse/transformers/pipelines/test_text_generation.py [hunk 2]
```diff
@@ -12,285 +12,236 @@
...
+                self.default_pipeline = Pipeline.create(**self.default_pipeline_kwargs)
+            return self.default_pipeline
         # return a pipeline with the given kwargs
-        return Pipeline.create(**kwargs)
+        # update the default pipeline kwargs
+        updated_kwargs = self.default_pipeline_kwargs.copy()
+        updated_kwargs.update(kwargs)
+        return Pipeline.create(**updated_kwargs)
+
+    def run_pipeline(
```
Comment (dbogunowicz) [inline tests/deepsparse/transformers/pipelines/test_text_generation.py:84]: Fixtures

Comment (dbogunowicz) [reply tests/deepsparse/transformers/pipelines/test_text_generation.py:84]: LRU cache

File: tests/deepsparse/transformers/pipelines/test_text_generation.py [hunk 3]
```diff
@@ -497,51 +380,52 @@ def _test_composition_same_session_ids(
...
+        assert numpy.allclose(cache_array_1, cache_array_2, atol=PRECISION)
         assert out_1.sequences[0] == out_2.sequences[0]
 
     def _test_output(
         self,
         output: "TextGenerationOutput",  # noqa F821
         cache_session: DecoderKVCache,
         torch_ground_truth: Tuple[numpy.ndarray, ...],
-        max_logits_difference_threshold: Optional[float] = None,
+        logits_threshold: Optional[float] = None,
```
Comment (dbogunowicz) [inline tests/deepsparse/transformers/pipelines/test_text_generation.py:428]: inspect the coverage

**Comment by dsikka** [comment]
Currently while running our `text_generation` tests, I install the `torch` and `transformers` deepsparse tags but I still get an error about `torchvision` not being installed. Dont want to create another necessarily but maybe we could encode the version somewhere?

File: tests/deepsparse/transformers/pipelines/configs/text_generation_gpt_neo.yaml [hunk 4]
```diff
@@ -0,0 +1,12 @@
...
+model_name: "roneneldan/TinyStories-1M"
+num_tokens_generate: 128
+prompt: "Didn't know what time it was, the lights were low\n I leaned back on my radio\n Some cat was layin' down some rock 'n' roll\n \"Lotta soul,\" he said\n Then the loud sound did seem to fade\n Came back like a slow voice on a wave of phase\n That weren't no DJ, that was hazy cosmic jive"
+has_bos_token: False
+logits_threshold: 24.7
+precision: 1e-3
+cache_management_type:
+    - "internal"
+    - "external"
+run_main_tests_only: False
```
Comment (dbogunowicz) [inline tests/deepsparse/transformers/pipelines/configs/text_generation_gpt_neo.yaml:11]: run_test: Optional[bool, List[str]] = ["test_ort_generation_after_kv_cache_has_been_filled", ...]
skipif

File: tests/deepsparse/transformers/pipelines/test_text_generation.py [hunk 5]
```diff
@@ -12,285 +12,249 @@
...
+#     has_bos_token: Whether the model has a bos token
+#     logits_threshold: The treshold for the max difference between the
+#                        actual and the expected logits in the situations
+#                        where they will not be able to match the ground
+#                        truth (e.g. when the DeepSparse pipeline is
+#                        running after the KV cache has been filled up).
+#                        This value is established
+#                        empirically for each combination of
+#                        prompt/pipeline/num_generated tokens.
+DIR_CONFIG_FILES = "tests/deepsparse/transformers/pipelines/configs"
```
Comment (dbogunowicz) [inline tests/deepsparse/transformers/pipelines/test_text_generation.py:44]: add a fixture for default tests to run

**Pull request closed without merging**